subworkflows/canine_octopus_module.cwl

```diff
@@ -0,0 +1,147 @@
+  disable_workflow: { type: 'boolean?', doc: "For when this workflow is wrapped into a larger workflow, you can use this value in the when statement to toggle the running of this workflow." }
+
+  calling_intervals: { type: 'File', doc: "YAML file contianing the intervals in which to perform variant calling." }
+  indexed_reference_fasta: { type: 'File', secondaryFiles: [{ pattern: ".fai", required: true }, { pattern: "^.dict", required: true }], doc: "Reference fasta with FAI index" }
+  input_reads: { type: 'File[]', secondaryFiles: [{ pattern: ".bai", required: false },{ pattern: "^.bai", required: false },{ pattern: ".crai", required: false },{ pattern: "^.crai", required: false }], doc: "BAM/CRAM files to be analysed." }
+  normal_sample_name: { type: 'string', doc: "BAM sample name of normal" }
+  output_basename: { type: 'string', doc: "String to use as basename for outputs." }
+  targets_file: { type: 'File?', doc: "For exome variant calling, this file contains the targets regions used in library preparation." }
+  cache_tar: { type: 'File?', doc: "Tarball of cache made by Octopus. Octopus will make this in the first run and it can be saved and reused." }
+
+  # Resource Control
+  octopus_ram: { type: 'int?', doc: "Maximum GB of RAM to allocate to Octopus." }
+  octopus_cpu: { type: 'int?', doc: "Number of CPUs to allocate to Octopus." }
+
+outputs:
+  octopus_all_vcf: { type: 'File', outputSource: bcftools_concat_sort_index/output }
+  octopus_pass_vcf: { type: 'File', outputSource: bcftools_filter_index/output }
+  octopus_all_vcf_stats: { type: 'File', outputSource: bcftools_stats_all/stats }
+  octopus_pass_vcf_stats: { type: 'File', outputSource: bcftools_stats_pass/stats }
+
+steps:
+  untar:
+    run: ../tools/untar.cwl
+    when: $(inputs.tarfile != null)
+    in:
+      tarfile: cache_tar
+      output_name:
+        source: disable_workflow # Hiding this here because I hate cavatica
+        valueFrom: ".cache"
+    out: [output]
+
+  calling_intervals_yaml_to_beds:
+    run: ../tools/calling_intervals_yaml_to_beds.cwl
+    in:
+      input_yaml: calling_intervals
+    out: [outputs]
+
+  octopus:
+    hints:
+      - class: 'sbg:AWSInstanceType'
+        value: c5.9xlarge
+    run: ../tools/octopus.cwl
+    scatter: [regions_file]
+    in:
+      premade_cache: untar/output
+      reference: indexed_reference_fasta
+      reads: input_reads
+      normal_sample: normal_sample_name
+      regions_file: calling_intervals_yaml_to_beds/outputs
+      caller:
+        valueFrom: "cancer"
+      max_reference_cache_footprint:
+        valueFrom: "4GB"
+      target_read_buffer_footprint:
+        valueFrom: "6GB"
+      ignore_unmapped_contigs:
+        valueFrom: $(1 == 1)
+      somatics_only:
+        valueFrom: $(1 == 1)
+      phasing_level:
+        valueFrom: "minimal"
+      legacy:
+        valueFrom: $(1 == 1)
+      annotations:
+        valueFrom: $(["AD","ADP","AF","SB"])
+      somatic_filter_expression:
```

Comment:
Hmmm, why set here instead of as an input with a default? Can be a "feature request" later if time is tight

Reply:
So my general philosophy here was: if its hardcoded in coyote, I hard code it like this. If its an input that's passed in, then I make it an input (with a default if possible). So yeah part of the porting process.